lib/connection.js

```diff
@@ -896,23 +898,29 @@ Connection.prototype.sobject = function(type) {
 /**
  * Get identity information of current user
  *
+ * @param {Object} [options] - Identity call options
+ * @param {Object} [options.headers] - Additional HTTP request headers sent in identity request
  * @param {Callback.<IdentityInfo>} [callback] - Callback function
  * @returns {Promise.<IdentityInfo>}
  */
-Connection.prototype.identity = function(callback) {
+Connection.prototype.identity = function(options, callback) {
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
   var self = this;
   var idUrl = this.userInfo && this.userInfo.url;
   return Promise.resolve(
     idUrl ?
     { identity: idUrl } :
-    this.request(this._baseUrl())
+    this.request(this._baseUrl(), options)
```

Comment:
Same case here and in line 923: url is a string, so I pass in options for request to retrieve headers and apply headers in API calls.